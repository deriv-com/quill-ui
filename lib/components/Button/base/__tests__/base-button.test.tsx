import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { LabelPairedPlaceholderMdRegularIcon } from "@deriv/quill-icons/LabelPaired";
import { Button } from "../../index";
import { ButtonProps } from "../../index";

const BasicButtonVariants: ButtonProps["variant"][] = [
    "primary",
    "secondary",
    "tertiary",
];
describe("Button component", () => {
    it("renders button text correctly", () => {
        render(<Button>Label</Button>);
        const buttonText = screen.getByText("Label");
        expect(buttonText).toBeInTheDocument();
    });

    it("Should handle onClick", async () => {
        const mockOnClick = jest.fn();
        const { getByRole } = render(
            <Button onClick={mockOnClick}>Label</Button>,
        );
        const button = getByRole("button");
        await userEvent.click(button);
        expect(mockOnClick).toHaveBeenCalled();
    });

    it("Should render BasicButton with Icon", () => {
        const { container } = render(
            <Button icon={<LabelPairedPlaceholderMdRegularIcon />}>
                Label
            </Button>,
        );
        expect(container).toMatchSnapshot();
    });

    BasicButtonVariants.forEach((variant) => {
        it(`Should render BasicButton with variant=${variant}`, () => {
            const { container } = render(
                <Button variant={variant}>Label</Button>,
            );
            expect(container).toMatchSnapshot();
        });
    });
});
