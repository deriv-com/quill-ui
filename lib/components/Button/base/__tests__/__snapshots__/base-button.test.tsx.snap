// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Should render BasicButton with Icon 1`] = `
<div>
  <button
    class="quill-button quill-button__size--md quill__color--primary-coral"
    data-state=""
  >
    <svg
      height="24"
      role="img"
      viewBox="0 0 14 24"
      width="14"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <path
          d="M2.75 5h.75c.25 0 .5.25.5.5 0 .281-.25.5-.5.5h-.75C1.781 6 1 6.813 1 7.75v.75c0 .281-.25.5-.5.5a.494.494 0 0 1-.5-.5v-.75C0 6.25 1.219 5 2.75 5M.5 10c.25 0 .5.25.5.5v3c0 .281-.25.5-.5.5a.494.494 0 0 1-.5-.5v-3c0-.25.219-.5.5-.5m13 0c.25 0 .5.25.5.5v3c0 .281-.25.5-.5.5a.494.494 0 0 1-.5-.5v-3c0-.25.219-.5.5-.5m0-1a.494.494 0 0 1-.5-.5v-.75C13 6.813 12.188 6 11.25 6h-.75a.494.494 0 0 1-.5-.5c0-.25.219-.5.5-.5h.75C12.75 5 14 6.25 14 7.75v.75c0 .281-.25.5-.5.5m.5 6.5v.75c0 1.531-1.25 2.75-2.75 2.75h-.75a.494.494 0 0 1-.5-.5c0-.25.219-.5.5-.5h.75c.938 0 1.75-.781 1.75-1.75v-.75c0-.25.219-.5.5-.5.25 0 .5.25.5.5m-13 0v.75c0 .969.781 1.75 1.75 1.75h.75c.25 0 .5.25.5.5 0 .281-.25.5-.5.5h-.75A2.734 2.734 0 0 1 0 16.25v-.75c0-.25.219-.5.5-.5.25 0 .5.25.5.5M5.5 19a.494.494 0 0 1-.5-.5c0-.25.219-.5.5-.5h3c.25 0 .5.25.5.5 0 .281-.25.5-.5.5zM5 5.5c0-.25.219-.5.5-.5h3c.25 0 .5.25.5.5 0 .281-.25.5-.5.5h-3a.494.494 0 0 1-.5-.5"
        />
      </g>
      <defs>
        <clippath
          id="28ec2eb9afe325f2205d77afa7750df1__a"
        >
          <path
            d="M0 0h14v24H0z"
          />
        </clippath>
      </defs>
    </svg>
    Label
  </button>
</div>
`;

exports[`Button component Should render BasicButton with variant=primary 1`] = `
<div>
  <button
    class="quill-button quill-button__size--md quill__color--primary-coral"
    data-state=""
  >
    Label
  </button>
</div>
`;

exports[`Button component Should render BasicButton with variant=secondary 1`] = `
<div>
  <button
    class="quill-button quill-button__size--md quill__color--secondary-coral"
    data-state=""
  >
    Label
  </button>
</div>
`;

exports[`Button component Should render BasicButton with variant=tertiary 1`] = `
<div>
  <button
    class="quill-button quill-button__size--md quill__color--tertiary-coral"
    data-state=""
  >
    Label
  </button>
</div>
`;
